src/core/server/http/integration_tests/default_route_provider_config.test.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import * as kbnTestServer from '../../../../test_utils/kbn_server';
+import { Root } from '../../root';
+
+const { startES } = kbnTestServer.createTestServers({
```

Comment:
This test should not be in the http service as it's application-level logic placed here https://github.com/elastic/kibana/pull/59694/files#diff-14bbd3d5a20ead019cd2d6d4ff234eb6R219
I'd like to discuss when we want to keep such logic. I think having something like `KibanaApp` sounds like a legitimate case here, and it adds a proper application/infrastructure level separation. WDYT?

Reply:
I can see this logic as somewhat analogous to the (poorly-named) RenderingService on the client-side which is responsible for composing and rendering the global UI tree. It's not really part of the base infrastructure & services used by plugins, but it is a feature of Core itself.

Having a component that acts more like a privileged plugin makes sense to me. This would be a component that exposes no APIs but implements Core features on top of the other Core services. This would certainly make testing easier since we can isolate the dependencies of this "application". My only hesitancy with `KibanaApp` is the name is a bit loaded already 😄. How about `CoreApp`?

Reply:
added `CoreApp`